src/Runtime.Base/src/System/Runtime/ThunkPool.cs

```diff
@@ -78,19 +78,19 @@ private class AllocatedBlock
         private AllocatedBlock _allocatedBlocks;
 
         // Helper functions to set/clear the lowest bit for ARM instruction pointers
-        private static IntPtr ClearThumbBit(IntPtr value)
+        private static unsafe IntPtr ClearThumbBit(IntPtr value)
         {
 #if ARM
             Debug.Assert(((nuint)value & 1) == 1);
-            value = (IntPtr)((nuint)value - 1);
+            value = new IntPtr((void *)((nuint)value - 1));
```

Comment:
I do not see a problem in the ClearThumbBit/SetThumbBit functions. The `(void*)` casts that you have added are superfluous.

Reply:
Would the `nuint` cause us to call the `long` overload of the IntPtr constructor that will do `(void*)checked((int)value)` in the body? This might be fixable by casting to `nint` instead of `nuint`.

Reply:
Oh I see - you right. I agree that casting to nint instead of nuint would be better fix.

Reply:
I'll try, I think nint will be better of course.